src/gap-skipper.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @file gap-skipper.js
+ */
+import Ranges from './ranges';
+import videojs from 'video.js';
+
+/**
+ * the gap skipper object handles all scenarios
+ * where the player runs into the end of a buffered
+ * region and there is a buffered region ahead.
+ * It then handles the skipping behavior.
+ *
+ * @class GapSkipper
+ */
+export default class GapSkipper {
+  /**
+   * Represents a GapSKipper object.
+   * @constructor
+   * @param {object} options an object that includes the tech and settings
+   */
+  constructor(options) {
+    if (!options.tech.options_.playerId) {
+      return;
+    }
+
+    this.player_ = videojs(options.tech.options_.playerId);
+    this.tech_ = options.tech;
+    this.consecutiveUpdates = 0;
+    this.lastRecordedTime = null;
+    this.timer_ = null;
+
+    if (options.debug) {
+      this.logger_ = videojs.log.bind(videojs, '<gap-skipper>');
+    }
+    this.logger_('<initialize>');
```

Comment:
Won't this output `<gap-skipper> <initialize>`? Are you planning on feeding your log statements to an XML parser? :)